driver-core/src/main/com/mongodb/operation/BulkWriteBatch.java

```diff
@@ -387,12 +387,23 @@ public void encode(final BsonWriter writer, final WriteRequest writeRequest, fin
                 writer.writeStartDocument();
                 writer.writeName("q");
                 getCodec(update.getFilter()).encode(writer, update.getFilter(), EncoderContext.builder().build());
-                writer.writeName("u");
 
-                if (update.getType() == WriteRequest.Type.UPDATE && update.getUpdate().isEmpty()) {
-                    throw new IllegalArgumentException("Invalid BSON document for an update");
+                if (update.getUpdate().isDocument()) {
```

Comment:
This looks like it can be simplified.  If the update is empty - either an empty document or pipeline its invalid and move the valid bson type checking into updateRequest.